cuqi/solver/_solver.py

```diff
@@ -320,6 +326,137 @@ def solve(self):
         return x, k
 
 
+class PCGLS:
+    """Conjugate Gradient method for least squares problems with preconditioning
+    
+    See Bjorck (1996) - Numerical Methods for Least Squares Problems. Pag 294.
+
+    Parameters
+    ----------
+    A : ndarray or callable f(x,*args).
+    b : ndarray.
+    x0 : ndarray. Initial guess.
+    P : ndarray. Preconditioner array in sparse format.
+    maxit : The maximum number of iterations.
+    tol : The numerical tolerance for convergence checks.
+    """    
+    def __init__(self, A, b, x0, P, maxit, tol=1e-6, shift=0):
+        self.A = A
+        self.b = b
+        self.x0 = x0
+        self.P = P
+        self.maxit = int(maxit)
+        self.tol = tol        
+        self.shift = shift
+        self.dim = len(x0)
+        if not callable(A):
+            self.explicitA = True
+        else:
+            self.explicitA = False
+        #
+        if self.dim < 1000:
```

Comment:
I would suggest using the same global config parameter for costly matrix computations: `cuqi.config.MAX_DIM_INV` instead of hard-coded 1000 (see here https://github.com/CUQI-DTU/CUQIpy/blob/main/cuqi/config.py#L4)